akka-stream/src/main/scala/akka/stream/impl/io/TcpListenStreamActor.scala

```diff
@@ -151,10 +155,17 @@ private[akka] class TcpListenStreamActor(localAddressPromise: Promise[InetSocket
     val (connected: Connected, connection: ActorRef) = incomingConnections.dequeueInputElement()
     val tcpStreamActor = context.watch(context.actorOf(TcpStreamActor.inboundProps(connection, halfClose, settings)))
     val processor = ActorProcessor[ByteString, ByteString](tcpStreamActor)
+
+    import scala.concurrent.duration._
+    val handler = (idleTimeout match {
+      case d: FiniteDuration ⇒ Flow[ByteString].via(Timeouts.idleTimeout(d))
```

Comment:
this should use the BidiStage version, this only checks for the input side, and does not consider long responses.